ui/app/helpers/utils/transactions.util.js

```diff
@@ -205,3 +205,53 @@ export function getBlockExplorerUrlForTx(networkId, hash, rpcPrefs = {}) {
   const prefix = getEtherscanNetworkPrefix(networkId);
   return `https://${prefix}etherscan.io/tx/${hash}`;
 }
+
+/**
+ * Returns a title for the given transaction category.
+ *
+ * This will throw an error if the transaction category is unrecognized and no default is provided.
+ * @param {function} t - The translation function
+ * @param {TRANSACTION_CATEGORIES[keyof TRANSACTION_CATEGORIES]} transactionCategory - The transaction category constant
+ * @param {string} [defaultTitle] - The default title to return if the transaction cateogory is not recognized.
+ * @returns {string} The transaction category title
+ */
+export function getTransactionCategoryTitle(
+  t,
+  transactionCategory,
+  defaultTitle,
+) {
+  switch (transactionCategory) {
+    case TRANSACTION_CATEGORIES.TOKEN_METHOD_TRANSFER: {
+      return t('transfer');
+    }
+    case TRANSACTION_CATEGORIES.TOKEN_METHOD_TRANSFER_FROM: {
+      return t('transferfrom');
+    }
+    case TRANSACTION_CATEGORIES.TOKEN_METHOD_APPROVE: {
+      return t('approve');
+    }
+    case TRANSACTION_CATEGORIES.SENT_ETHER: {
+      return t('sentEther');
+    }
+    case TRANSACTION_CATEGORIES.CONTRACT_INTERACTION: {
+      return t('contractInteraction');
+    }
+    case TRANSACTION_CATEGORIES.DEPLOY_CONTRACT: {
+      return t('contractDeployment');
+    }
+    case TRANSACTION_CATEGORIES.SWAP: {
+      return t('swap');
+    }
+    case TRANSACTION_CATEGORIES.SWAP_APPROVAL: {
+      return t('swapApproval');
+    }
+    default: {
+      if (defaultTitle) {
+        return defaultTitle;
+      }
+      throw new Error(
+        `Unrecognized transaction category: ${transactionCategory}`,
```

Comment:
maybe weird that we dont hit this validation if the defaultTitle is present

Reply:
I meant the default case to imply that it was OK for the category to not be matched. That's how it's used in `useTransactionDisplayData` for example - there was a fallback present.

But now that you mention it... that doesn't seal ideal, I agree. Maybe we can remove the need for that fallback case. I'll look into that.

Reply:
This was addressed in https://github.com/MetaMask/metamask-extension/pull/10391/commits/a8ab8e8f99ca273a9117b93f1e9348e7add71952

The default case has been replaced by another condition in the one case it was relied upon.